### PR TITLE
CIV-10053 Invalid HwF

### PR DESCRIPTION
**Pull request by ElifDenizEsea**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-10053

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-R2-CUI-nonprod.json [hunk 1]
```diff
@@ -627,6 +627,18 @@
       }
     ]
   },
+  {
+    "CaseTypeID": "CIVIL",
+    "CaseEventID": "INVALID_HWF_REFERENCE",
+    "AccessControl": [
+      {
+        "UserRoles": [
+          "caseworker-civil-admin"
```
Comment (portizdev) [inline ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-R2-CUI-nonprod.json:636]: As we discussed yesterday,we only need "cui-admin-profile" here.

File: ccd-definition/AuthorisationCaseField/AuthorisationCaseField-R2-CUI-nonprod.json [hunk 2]
```diff
@@ -2449,6 +2449,21 @@
       }
     ]
   },
+  {
+    "CaseTypeID": "CIVIL",
+    "CaseFieldID": "helpWithFeesReferenceNumber",
```
Comment (portizdev) [inline ccd-definition/AuthorisationCaseField/AuthorisationCaseField-R2-CUI-nonprod.json:2454]: We already have this field in ccd, there is no need to add it again.
We just need to change the authorized roles to the "hearingHelpFeesReferenceNumber" field. (You can check other hwf prs)

File: ccd-definition/CaseEvent/User/UserEventsLRspec-R2-CUI-nonprod.json [hunk 3]
```diff
@@ -36,5 +36,21 @@
     "RetriesTimeoutAboutToStartEvent": 0,
     "RetriesTimeoutURLAboutToSubmitEvent": 0,
     "RetriesTimeoutURLSubmittedEvent": 0
+  },
+  {
+    "CaseTypeID": "CIVIL",
+    "ID": "INVALID_HWF_REFERENCE",
+    "Name": "Invalid HwF Reference",
+    "Description": "Invalid HwF Reference",
+    "PreConditionState(s)": "*",
```
Comment (portizdev) [inline ccd-definition/CaseEvent/User/UserEventsLRspec-R2-CUI-nonprod.json:45]: Add the status for pre-condition, I think they are PENDING_CASE_ISSUED and HEARING_READINESS

File: ccd-definition/CaseEventToFields/InvalidHwFReference-R2-CUI-nonprod.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
+[
+  {
+    "CaseTypeID": "CIVIL",
+    "CaseEventID": "INVALID_HWF_REFERENCE",
+    "CaseFieldID": "caseTitleWithHwFReference",
```
Comment (portizdev) [inline ccd-definition/CaseEventToFields/InvalidHwFReference-R2-CUI-nonprod.json:5]: You need to adapt the title to use the HWF reference number of claim fee or hearing fee depending on the type. You can check other PRs.

**Review by portizdev — CHANGES_REQUESTED** [review]

File: ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-R2-CUI-nonprod.json (hunk 1 above)
Comment (ElifDenizEsea) [reply ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-R2-CUI-nonprod.json:636]: done

File: ccd-definition/AuthorisationCaseField/AuthorisationCaseField-R2-CUI-nonprod.json (hunk 2 above)
Comment (ElifDenizEsea) [reply ccd-definition/AuthorisationCaseField/AuthorisationCaseField-R2-CUI-nonprod.json:2454]: done

File: ccd-definition/CaseEvent/User/UserEventsLRspec-R2-CUI-nonprod.json (hunk 3 above)
Comment (ElifDenizEsea) [reply ccd-definition/CaseEvent/User/UserEventsLRspec-R2-CUI-nonprod.json:45]: done

File: ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-R2-CUI-nonprod.json [hunk 5]
```diff
@@ -627,6 +627,18 @@
       }
     ]
   },
+  {
+    "CaseTypeID": "CIVIL",
+    "CaseEventID": "INVALID_HWF_REFERENCE",
+    "AccessControl": [
+      {
```
Comment (manish14836) [inline ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-R2-CUI-nonprod.json:634]: This should be part of AuthorisationCaseEvent-HWF-R2-CUI-nonprod.json file

File: ccd-definition/CaseField/CaseField-CUI-R2-nonprod.json [hunk 6]
```diff
@@ -33,6 +33,40 @@
...
+    "Searchable": "N",
+    "Publish": "Y"
+  },
+  {
+    "CaseTypeID": "CIVIL",
+    "ID": "caseTitleWithHwFReference",
+    "Label": "# ${applicant1.partyName} Vs ${respondent1.partyName}",
+    "FieldType": "Label",
+    "SecurityClassification": "Public",
+    "Searchable": "N"
```
Comment (manish14836) [inline ccd-definition/CaseField/CaseField-CUI-R2-nonprod.json:51]: Move these to HWF related file CaseField-HWF-R2-CUI-nonprod.json. Also please check others PR how these are define and follow same to make these consistent

**Review by manish14836 — CHANGES_REQUESTED** [review]

File: ccd-definition/CaseEvent/User/UserEvents-HWF-R2-CUI-nonprod.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
+[
+  {
+    "CaseTypeID": "CIVIL",
+    "ID": "INVALID_HWF_REFERENCE",
+    "Name": "Invalid HwF Reference",
+    "Description": "Invalid HwF Reference",
+    "PreConditionState(s)": "HEARING_READINESS;PENDING_CASE_ISSUED",
+    "PostConditionState": "*",
+    "SecurityClassification": "Public",
+    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/about-to-submit",
```
Comment (portizdev) [inline ccd-definition/CaseEvent/User/UserEvents-HWF-R2-CUI-nonprod.json:10]: add about to start event

**Review by jeswanth-hmcts — DISMISSED** [review]

**Review by portizdev — APPROVED** [review]

File: ccd-definition/CaseEvent/User/UserEvents-HWF-R2-CUI-nonprod.json [hunk 8]
```diff
@@ -7,6 +24,7 @@
     "PreConditionState(s)": "PENDING_CASE_ISSUED;HEARING_READINESS",
     "PostConditionState": "*",
     "SecurityClassification": "Public",
+    "CallBackURLAboutToStartEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/about-to-start",
```
Comment (neeta-hmcts) [inline ccd-definition/CaseEvent/User/UserEvents-HWF-R2-CUI-nonprod.json:27]: move this to your event

**Review by neeta-hmcts — APPROVED** [review]

**Review by manish14836 — DISMISSED** [review]

**Review by portizdev — APPROVED** [review]

File: ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-HWF-R2-CUI-nonprod.json [hunk 9]
```diff
@@ -13,19 +13,7 @@
   },
   {
     "CaseTypeID": "CIVIL",
-    "CaseEventID": "PARTIAL_REMISSION_HWF_GRANTED",
```
Comment (jeswanth-hmcts) [inline ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-HWF-R2-CUI-nonprod.json:16]: is PARTIAL_REMISSION_HWF_GRANTED not needed?

File: ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-HWF-R2-CUI-nonprod.json [hunk 10]
```diff
@@ -13,19 +13,7 @@
...
-        "UserRoles": [
-          "cui-admin-profile"
-        ],
-        "CRUD": "CRU"
-      }
-    ]
-  },
-   {
-    "CaseTypeID": "CIVIL",
-    "CaseEventID": "MORE_INFORMATION_HWF",
```
Comment (jeswanth-hmcts) [inline ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-HWF-R2-CUI-nonprod.json:28]: even MORE_INFORMATION_HWF is not  needed , check is it replaced while resolving the conflicts

**Review by neeta-hmcts — APPROVED** [review]

**Review by jeswanth-hmcts — DISMISSED** [review]

**Pull request merged**